### PR TITLE
 Protect the ESProvisioning setting behind a mutex

### LINKED ISSUE #1983: fatal error: concurrent map read and map write
**Describe the bug**
The error happens sometimes, i guess some process triggers that.
Please ignore `This field was missing!!!!!!!!` as it is our internal message.

Please find attached file with surrouding logs 

[On_demand_report_2022-07-05T10_54_28.180Z_d7538140-fc50-11ec-84ba-cf7f04fb0981.csv](https://github.com/jaegertracing/jaeger-operator/files/9045750/On_demand_report_2022-07-05T10_54_28.180Z_d7538140-fc50-11ec-84ba-cf7f04fb0981.csv)

### PR DESCRIPTION
**Pull request by iblancasa**
## Which problem is this PR solving?
- Continuation of #2278 to fix #1983

## Short description of the changes
- Protect the ESProvisioning setting behind a mutex

### CODE REVIEW

File: pkg/autodetect/main_test.go [hunk 1]
```diff
@@ -88,7 +85,7 @@ func TestStartContinuesInBackground(t *testing.T) {
 	select {
 	case <-done:
 		assert.False(t, viper.GetBool("auth-delegator-available"))
-	case <-time.After(5 * time.Second):
+	case <-time.After(1 * time.Second):
```
Comment (frzifus) [inline pkg/autodetect/main_test.go:88]: Why this change?

File: pkg/autodetect/main_test.go [hunk 2]
```diff
@@ -100,7 +97,7 @@ func TestStartContinuesInBackground(t *testing.T) {
 			if viper.GetBool("auth-delegator-available") {
 				break
 			}
-			time.Sleep(10 * time.Millisecond)
+			time.Sleep(500 * time.Millisecond)
```
Comment (frzifus) [inline pkg/autodetect/main_test.go:100]: same here?

File: pkg/autodetect/operatorconfig.go [hunk 3]
```diff
@@ -24,6 +24,21 @@ func (p Platform) String() string {
 	return [...]string{"Kubernetes", "OpenShift"}[p]
 }
 
+// ESOperatorIntegration holds the if the ES Operator integration is enabled.
+type ESOperatorIntegration int
+
+const (
+	// ESOperatorIntegrationYes represents the ES Operator integration is enabled.
+	ESOperatorIntegrationYes ESOperatorIntegration = iota
```
Comment (frzifus) [inline pkg/autodetect/operatorconfig.go:32]: Does it make sense to start here with an `ESOperatorIntegrationUnknown` value?

File: pkg/autodetect/main_test.go (hunk 1 above)
Comment (iblancasa) [reply pkg/autodetect/main_test.go:88]: The test is now more resilient. We don't need to wait all that time.

File: pkg/autodetect/operatorconfig.go (hunk 3 above)
Comment (iblancasa) [reply pkg/autodetect/operatorconfig.go:32]: I don't think it is needed

**Review by rubenvp8510 — APPROVED** [review]

**Pull request merged**